File: prototypes/01-priyas-first-honest-morning-Prototype/shared/nav.js
/* ============================================================================
   nav.js — left sidebar navigation + linear scenario routing (prototype)
   Desktop-only left sidebar, used by all authenticated views
   (Dashboard, Transactions, Insights, Copilot).
   Commitments is NOT a sidebar item (2026-07-09) — reached only via the
   Dashboard's "+ Add a commitment" button, matching FR-6.4's 4-screen nav.
   ============================================================================ */

/* Linear golden-path order (page file names). Dashboard is the landing page
   immediately after Upload; Transactions is reached from the Dashboard's
   "View All Transactions" CTA or the sidebar, not automatically post-upload. */
const SCENARIO_ORDER = [
  '01.1-register.html',
  '01.2-login.html',
  '01.3-statement-upload.html',
  '01.5-dashboard.html',
  '01.4-transactions-table.html',
  '01.6-ai-insights-recommendations.html',
  '01.7-copilot-chat.html'
];

/** Navigate to the next page in the golden path. */
function goNext(currentPage) {
  const i = SCENARIO_ORDER.indexOf(currentPage);
  if (i > -1 && i < SCENARIO_ORDER.length - 1) {
    window.location.href = SCENARIO_ORDER[i + 1];
  }
}

/** Navigate directly to a page file. */
function goTo(page) { window.location.href = page; }

/**
 * Render the persistent LEFT sidebar nav into #side-nav-slot.
 * Nav order per ux-spec-mvp.md FR-6.4 / epics.md: Dashboard, Transactions, Insights, Copilot.
 * Commitments is intentionally not a sidebar item — reached via the Dashboard's
 * "+ Add a commitment" button (02.1-commitments-management.html renders this same
 * sidebar with no active tab, since it isn't one of the 4 primary destinations).
 * activeTab: 'dashboard' | 'transactions' | 'insights' | 'copilot'
 */
function renderSideNav(activeTab) {
  var slot = document.getElementById('side-nav-slot');
  if (!slot) return;
  var tabs = [
    { key: 'dashboard',    label: 'Dashboard',    icon: '📊', page: '01.5-dashboard.html' },
    { key: 'transactions', label: 'Transactions', icon: '📄', page: '01.4-transactions-table.html' },
    { key: 'insights',     label: 'Insights',     icon: '💡', page: '01.6-ai-insights-recommendations.html' },
    { key: 'copilot',      label: 'Copilot Chat', icon: '💬', page: '01.7-copilot-chat.html' }
  ];
  var items = tabs.map(function (t) {
    var active = t.key === activeTab;
    return '<button class="side-nav__tab' + (active ? ' is-active' : '') + '"' +
      (active ? ' aria-current="page"' : '') + ' onclick="goTo(\'' + t.page + '\')">' +
      '<span class="ico" aria-hidden="true">' + t.icon + '</span>' +
      '<span class="label">' + t.label + '</span></button>';
  }).join('');
  // Logout pinned to the bottom of the rail
  items += '<button class="side-nav__tab side-nav__logout" onclick="Auth.logout()">' +
    '<span class="ico" aria-hidden="true">⏻</span><span class="label">Log out</span></button>';
  slot.innerHTML = items;
}

window.goNext = goNext;
window.goTo = goTo;
window.renderSideNav = renderSideNav;

File: finance_app/components/nav.py
"""Persistent left sidebar navigation.

Mirrors the prototype's ``nav.js``/``renderSideNav`` exactly:
- 4 tabs: Dashboard, Transactions, Insights, Copilot Chat
- Active tab gets ``.is-active`` + ``aria-current="page"``
- Log out pinned at the bottom via ``.side-nav__logout``
- Commitments is intentionally absent (FR-6.4 / nav.js comment: reached via Dashboard)

Usage in a page::

    from finance_app.components.nav import side_nav
    ...
    rx.el.div(
        side_nav("dashboard"),
        rx.el.main(..., class_name="has-sidenav"),
    )

``active`` must be one of: ``"dashboard"`` | ``"transactions"`` | ``"insights"`` | ``"copilot"``.
"""
from __future__ import annotations

import reflex as rx

from finance_app.state.auth_state import AuthState

_TABS = [
    ("dashboard",    "Dashboard",    "📊", "/dashboard"),
    ("transactions", "Transactions", "📄", "/transactions"),
    ("insights",     "Insights",     "💡", "/insights"),
    ("copilot",      "Copilot Chat", "💬", "/copilot"),
]


def _tab(key: str, label: str, icon: str, href: str, active: str) -> rx.Component:
    is_active = key == active
    return rx.el.a(
        rx.el.span(icon, class_name="ico", aria_hidden="true"),
        rx.el.span(label, class_name="label"),
        href=href,
        class_name="side-nav__tab" + (" is-active" if is_active else ""),
        aria_current="page" if is_active else None,
    )


def side_nav(active: str) -> rx.Component:
    """Render the fixed left sidebar with the 4 primary nav tabs + logout.

    ``active``: the tab key for the current page (sets ``.is-active`` + ``aria-current``).
    """
    tabs = [_tab(key, label, icon, href, active) for key, label, icon, href in _TABS]
    logout_btn = rx.el.button(
        rx.el.span("⏻", class_name="ico", aria_hidden="true"),
        rx.el.span("Log out", class_name="label"),
        on_click=AuthState.logout,
        class_name="side-nav__tab side-nav__logout",
        type="button",
    )
    return rx.el.nav(
        *tabs,
        logout_btn,
        class_name="side-nav",
        aria_label="Primary",
    )
